hnn_core/network.py

```diff
@@ -349,10 +349,15 @@ def add_evoked_drive(self, name, *, mu, sigma, numspikes,
             Standard deviation of event time distribution
         numspikes : int
             Number of spikes at each target cell
-        sync_within_trial : bool
-            If True, the target cells receive each numspike synchronously. By
-            default (False), spike times arriving at each target cell are
-            sampled independently using the Gaussian parameteres (mu, sigma).
+        numsources : int | None
+            The number of sources (i.e., ArtificialCell objects) that each
+            contribute an iid-sampled synaptic spike to the network according
+            to the Gaussian time distribution (mu, sigma). If None, a source
+            gets assigned to each of the self.n_cells simulated cells in the
+            network with 1-to-1 connectivity. Otherwise, sources are assigned
+            with all-to-all connectivity. If you wish to synchronize the timing
+            of this evoked drive across the network in a given trial with one
+            spike, set numsources=1.
```

Comment:
I like this if it was indeed the intended function. Perhaps worth further comments from @cjayb and @ntolley though

One thing this docstring brings to the fore is that in our `add_xxx_drive` methods do two things: 1) add a artificial cell population with either 1 cell or n_cells or in this case numsources. Each of the cells has an independently sampled distribution of spikes associated with it 2) connect the artificial cells to the network cells using either 1-to-1 connectivity or all-to-all. 

It might be worth clarifying this in the docstring of all the methods

Reply:
I think this is a clear improvement. There's still something funky about `None` being synonymous to ' `n_drive_cells`-to-`n_cells` ', but I can't come up with a better suggestion... I think getting rid of `sync_...` and `repeats` is a worthy cause though!

Reply:
agreed, it's a little funky. I keep thinking about going towards an API that's close to @cjayb 's original API suggestion in the google docs. Something like:

```py
drive = Drive(n_cells=100, name='thalamic-evoked')
drive.add_distribution(spike_func)
network.add_cell_type(drive)
network.add_connection('thalamic-evoked', 'L5_Pyramidal', ...)
```

But maybe not worth the overhaul effort except for cleaning up internally as much as possible. c.f.: https://github.com/jonescompneurolab/hnn-core/pull/369 

For now, perhaps instead of `None`, it might be clearer if you say:

```py
n_drive_cells : int | 'n_cells'
If 'n_cells' ... blah
```

Reply:
How do you feel about a string argument instead of `None`? Something like `n_drive_cells='equal'`

Reply:
My vote is for `'n_cells'`, though I'm cool with `'equal'` as well. Out of curiosity, what's convention here? For some reason I was under the impression that using `None` was best practice for default arguments that require more complex logic or that reference an object attribute (e.g., in this case, `self.n_cells`).

Reply:
not really, `None` is mostly for mutable data types such as lists which should never be set in the function. Otherwise you can start seeing unexpected things:

```py
def myfunc(x=[1, 2, 3]):
   x.append(x[-1] + 1)
   print(x[-1])

myfunc()
myfunc()
```

Expected output is:

```sh
4
4
```

but you get:

```sh
4
5
```

That's why the idiom is:

```py
def myfunc(x=None):
   if x is None:
       x = list()
```

Reply:
Got it, thanks @jasmainak! Note that I changed the default for `add_evoked_drive()` to `n_drive_cells='n_cells'` in https://github.com/jonescompneurolab/hnn-core/pull/383/commits/501d357ba4003ba6d2cd175d4c1f589f132c95cf.